cmd/krew/cmd/update.go

```diff
@@ -132,6 +145,9 @@ func ensureIndexUpdated(_ *cobra.Command, _ []string) error {
 	// TODO(chriskim06) consider commenting this out when refactoring for custom indexes
 	showUpdatedPlugins(os.Stderr, preUpdateIndex, posUpdateIndex, installedPlugins)
 
+	if len(failed) != 0 {
+		return errors.Errorf("failed to update the following indexes: %s\n", strings.Join(failed, ", "))
```

Comment:
Something else we do in install.go is that we collect the first error as `returnErr` and wrap+return it.

https://github.com/kubernetes-sigs/krew/blob/b2b8a40555208cf5ff2dcb017befca0edbb54480/cmd/krew/cmd/install.go#L161-L163

https://github.com/kubernetes-sigs/krew/blob/b2b8a40555208cf5ff2dcb017befca0edbb54480/cmd/krew/cmd/install.go#L178-L180

The reason for this is that when you wrap+return errors, when executed with `-v 1` or above, we the retain ability to print full stack trace (in this case, at least for 1 error which is good enough).